contrib/platform/src/com/sun/jna/platform/mac/CoreFoundationUtil.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2019 Daniel Widdis
+ *
+ * The contents of this file is dual-licensed under 2
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ *
+ * You can freely decide which license you want to apply to
+ * the project.
+ *
+ * You may obtain a copy of the LGPL License at:
+ *
+ * http://www.gnu.org/licenses/licenses.html
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ *
+ * You may obtain a copy of the Apache License at:
+ *
+ * http://www.apache.org/licenses/
+ *
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.mac;
+
+import static com.sun.jna.platform.mac.CoreFoundation.kCFStringEncodingUTF8;
+
+import java.util.Collection;
+
+import com.sun.jna.Memory;
+import com.sun.jna.platform.mac.CoreFoundation.CFAllocatorRef;
+import com.sun.jna.platform.mac.CoreFoundation.CFNumberType;
+import com.sun.jna.platform.mac.CoreFoundation.CFTypeRef;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
+
+/**
+ * Provides utilities for Core Foundations
+ */
+public class CoreFoundationUtil {
+
+    private static final CoreFoundation CF = CoreFoundation.INSTANCE;
+
+    public static final CFAllocatorRef ALLOCATOR = CF.CFAllocatorGetDefault();
+
+    private CoreFoundationUtil() {
+    }
+
+    /**
+     * Convert a reference to a Core Foundations LongLong into its {@code long}
+     *
+     * @param theLong
+     *            The pointer to a 64-bit integer
+     * @return The corresponding {@code long}
+     */
+    public static long cfPointerToLong(CFTypeRef theLong) {
+        LongByReference lbr = new LongByReference();
+        CF.CFNumberGetValue(theLong, CFNumberType.kCFNumberLongLongType.ordinal(), lbr);
+        return lbr.getValue();
+    }
+
+    /**
+     * Convert a reference to a Core Foundations Int into its {@code int}
+     *
+     * @param p
+     *            The pointer to an integer
+     * @return The corresponding {@code int}
+     */
+    public static int cfPointerToInt(CFTypeRef p) {
+        IntByReference ibr = new IntByReference();
+        CF.CFNumberGetValue(p, CFNumberType.kCFNumberIntType.ordinal(), ibr);
+        return ibr.getValue();
+    }
+
+    /**
+     * Convert a reference to a Core Foundations Boolean into its {@code boolean}
+     *
+     * @param cfBoolean
+     *            The pointer to a boolean
+     * @return The corresponding {@code boolean}
+     */
+    public static boolean cfPointerToBoolean(CFTypeRef cfBoolean) {
+        return CF.CFBooleanGetValue(cfBoolean);
```

Comment:
What is the size of `Boolean` on Mac OS X? If it is not 4, this might blow.

Reply:
Boolean maps to `char` but this is another instance of the type safety question above regarding the return value of `CFDictionaryGetValue()`.  If the dictionary entry is a boolean, the return will be a `CFBooleanRef` which is one of two constant values (`kCFBooleanTrue` or `kCFBooleanFalse`), both of which would be safe here.  We can narrow the typing here at the cost of having to cast the return from `CFDictionaryGetValue()`.

Reply:
If the native `boolean` is a typedef for the native `char`, `CFBooleanGetValue` needs to have a return type of `byte`, the java `boolean` is mapped to a native `int`.

Reply:
I think the source code location of this comment is confusing the discussion.  I didn't realize that you're actually talking about the return type of the `CFBooleanGetValue()` function mapped in the other class.  I'll dig into this more to confirm.

Reply:
Looks like a `byte` return type applies to all the existing boolean returns so I'll update them.  The value is always (byte) 1 or (byte) 0.